src/go/rpk/pkg/cli/profile/create.go

```diff
@@ -269,14 +323,50 @@ func createCloudProfile(ctx context.Context, y *config.RpkYaml, cfg *config.Conf
 		return CloudClusterOutputs{}, err
 	}
 	if expired {
-		return CloudClusterOutputs{}, fmt.Errorf("token for %q has expired, please login again", y.CurrentCloudAuth)
+		return CloudClusterOutputs{}, errors.New("current cloud auth has expired, please re-login with 'rpk cloud login'")
 	}
 	cl := cloudapi.NewClient(overrides.CloudAPIURL, a.AuthToken, httpapi.ReqTimeout(10*time.Second))
 
-	if clusterID == "prompt" {
+	if clusterIDOrName == "prompt" {
 		return PromptCloudClusterProfile(ctx, cl)
 	}
 
+	var clusterID string
+	_, err = xid.FromString(clusterIDOrName)
```

Comment:
We may want to try the OrgNamespacesClusters block if VirtualCluster and Cluster both fail with this xid.